Apps/phdocker/docker_v3_connector.py

```diff
@@ -0,0 +1,1082 @@
+                    url,
+                    verify=config.get("verify_server_cert", False),
+                    json=k_data
+                )
+            else:
+                r = request_func(
+                    url,
+                    verify=config.get('verify_server_cert', False),
+                    **kwargs
+                )
+        except requests.exceptions.InvalidURL:
+            error_message = "Error connecting to server. Invalid URL %s" % (url)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, error_message), resp_json)
+        except requests.exceptions.ConnectionError:
+            error_message = "Error connecting to server. Connection Refused from the Server for %s" % (url)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, error_message), resp_json)
+        except requests.exceptions.InvalidSchema:
+            error_message = "Error connecting to server. No connection adapters were found for %s" % (url)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, error_message), resp_json)
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            return RetVal(
+                action_result.set_status(
+                    phantom.APP_ERROR,
+                    "Error Connecting to server. {0}".format(err)
+                ), resp_json
+            )
+
+        # Below line is commented out as it sets the status_code explicitly to 200
+        # This creates a problem in the error message when the original response code is something else than 200(like 500)
+        # r.status_code = 200
+        return self._process_response(r, action_result)
+
+    def _cleanup_row_values(self, row):
+        # The MySQL column values is supposed
+        # to be a bytearray as opposed to a string
+        return {k: v.decode('utf-8')
+                if type(v) == bytearray else v for k, v in row.items()}
+
+    def _handle_test_connectivity(self, param):
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        self.save_progress("Obtaining the version of the docker host")
+        # make rest call
+        ret_val, response = self._make_rest_call('/version', action_result)
+        if phantom.is_fail(ret_val):
+            self.save_progress("Test Connectivity Failed")
+            return action_result.get_status()
+
+        try:
+            res = json.dumps(response)
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            self.debug_print("Error occurred while parsing response. {}".format(err))
+
+        indices = [i + 1 for i, elem in enumerate(res) if elem == ',']
+        indices.insert(0, 0)
+        indices.insert(len(indices), len(res))
+        for i in range(len(indices) - 1):
+            self.save_progress(res[indices[i]:indices[i + 1]])
+        self.save_progress(self._base_url)
+        # action_result.add_data(response)
```

Comment:
You might want to remove the commented code - line 324 has unbalanced parentheses.

Reply:
Fixed